### ISSUE TITLE
nextcloud-client segmentation fault on nixos-unstable

### ISSUE THREAD

**Issue by lycheese** [NixOS/nixpkgs#119029]
**Describe the bug**
After #118084 was merged nextcloud-client no longer fails to build, but now it crashes with SIGSEGV.

**To Reproduce**
1. `nix-shell -p nextcloud-client`
2. `nextcloud`

**Expected behavior**
nextcloud-client starts and works.

**Additional context**
<details><summary>gdb core dump</summary>

```
warning: Can't open file /memfd:JSGCHeap:QtQml (deleted) during file-backed mapping note processing
warning: Can't open file /memfd:JSVMStack:QtQml (deleted) during file-backed mapping note processing
warning: Can't open file /SYSV51260028 (deleted) during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file /memfd:xshmfence (deleted) during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
warning: Can't open file /run/nscd/dbe8hReO (deleted) during file-backed mapping note processing
warning: Can't open file anon_inode:i915.gem which was expanded to anon_inode:i915.gem during file-backed mapping note processing
[New LWP 76379]
[New LWP 76363]
[New LWP 76369]
[New LWP 76371]
[New LWP 76374]
[New LWP 76370]
[New LWP 76381]
[New LWP 76375]
[New LWP 76373]
[New LWP 76372]
[New LWP 76380]
[New LWP 76377]
[New LWP 76376]
[New LWP 76378]
[New LWP 76362]
[New LWP 76382]
Core was generated by `/run/current-system/sw/bin/nextcloud'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f7ba9a6d590 in ?? ()
[Current thread is 1 (LWP 76379)]

```
</details>

**Notify maintainers**
@caugner 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.27, NixOS, 21.05.20210408.9e377a6 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210326_dd77f71`
 - nixpkgs: `/nix/store/mmws9c6185xn3wvlzxa4a4h68x9q26yw-source`
```
nixpkgs rev: 9e377a6ce42dccd9b624ae4ce8f978dc892ba0e2

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- nextcloud-client
# a list of nixos modules affected by the problem
module:
```

**Comment by jonringer** [comment]
I'm able to run this on unstable, and have the gui launch. I did notice
```
/run/current-system/sw/bin/nextcloud
```

You are likely running nextcloud from your system, and not from nix-shell, try `nix-shell --pure -p nextcloud-client --run nextcloud`

**Comment by lycheese** [comment]
Still the same.
Output of `nix-shell --pure -p nextcloud-client --run nextcloud`:
```
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/run/user/1000/runtime-lycheese'
/tmp/nix-shell-1814-0/rc: line 1:  3909 Segmentation fault      (core dumped) nextcloud
```

But if it works on your end, it must be something in my config. Not sure what could cause this though. Is there a way to prevent a nix-shell from accessing /run?

**Comment by jonringer** [comment]
> Is there a way to prevent a nix-shell from accessing /run?

I think that's okay, and probably not affect the behavior.

> Or force rebuilding of all dependencies?

`nix-build -E 'import <nixpkgs> {}; pkgs.nextcloud-client' --check`  for nix <2.4
`nix build .#nextclout-client --rebuild` nix >=2.4

**Comment by lycheese** [comment]
I am on nix version `2.4pre20210326_dd77f71` and `nix build .#nextclout-client --rebuild` is giving me an error that my path is not a flake. I am using flakes for my system configuration and do not have any nix channels, if that is relevant.

**Comment by lycheese** [comment]
Since it works on your end, should we just close this issue? I don't really want to take up your time with fixing my configuration…

**Comment by jonringer** [comment]
> giving me an error that my path is not a flake.

oh right, it would be `nix build nixpkgs#nextcloud-client --rebuild`, when outside of the tree

**Comment by lycheese** [comment]
> > giving me an error that my path is not a flake.
> 
> oh right, it would be `nix build nixpkgs#nextcloud-client --rebuild`, when outside of the tree

The command worked, but when following the result link and executing bin/nextcloud it still throws a segmentation error.

**Comment by Ekleog** [comment]
FWIW it's segfaulting for me too, on nixpkgs f5e8bdd with nextcloud-client installed in environment.systemPackages, though I don't have the first line your logs show and I directly get a segfault. Haven't had the time to investigate yet though, but I don't think this should be closed yet. Maybe it's related to not using KDE as a DE?

**Comment by lycheese** [comment]
I managed to fix it on my side.

After confirming that I managed to run it successfully on a new install with the same versions I suspected it had something to do with remaining configuration.

Running `nix-shell --pure -p strace nextcloud --run 'strace -fte trace=file nextcloud'` showed that immediately before segfaulting it tried to open a file in an existing directory. After moving the `Nextcloud` folders in `~/.cache`, `~/.config` and `~/.local/share` the nextcloud client started normally.

@Ekleog does moving/deleting those directories work for you as well?

**Comment by turion** [comment]
I had the same issue. Deleting `~/.cache/Nextcloud` fixed it for me. it contains files like these:

```
$ ls ~/.cache/Nextcloud/qmlcache/
01567610c4e844cecc63495b5844e454d8fc94ee.qmlc  4da21d846270ee9247c9b9821dfa3afbbbb52097.qmlc [...]
```

So likely triggering some Qt version mismatch?

**Comment by das-g** [comment]
> Running `nix-shell --pure -p strace nextcloud --run 'strace -fte trace=file nextcloud'` showed [...]

That should probably be

> Running `nix-shell --pure -p strace nextcloud-client --run 'strace -fte trace=file nextcloud'` showed [...]

(package `nextcloud-client`, not `nextcloud`)

**Comment by lycheese** [comment]
> > Running `nix-shell --pure -p strace nextcloud --run 'strace -fte trace=file nextcloud'` showed [...]
> 
> That should probably be
> 
> > Running `nix-shell --pure -p strace nextcloud-client --run 'strace -fte trace=file nextcloud'` showed [...]
> 
> (package `nextcloud-client`, not `nextcloud`)

You are totally right. Fixed.

**Comment by lycheese** [comment]
@Ekleog Are you still having this problem? If not, I would like to close this issue.

**Comment by Ekleog** [comment]
Oh. I'm sorry, I wanted to actually dive some more into this and figure out a way to make the migration to 21.05 more seamless, but it looks like I'm way late and 21.05 already forked anyway so it's too late to add this kind of workarounds to the build…

I can confirm that the `rm -r ~/.cache/Nextcloud` workaround worked for me too, so let's close as you were suggesting.

Thank you for the solution!

**Comment by aviallon** [comment]
I recently had the same issue, with the same fix, on 22.05.
Maybe https://github.com/NixOS/nixpkgs/commit/788119f78101a10b2c58de0690c23572ac348076 got reverted?
